File: snippets/fsharp/System/Convert/ToString/tostring_string1.fs
module tostring_string1

// <Snippet2>
open System

let article = "An"
let noun = "apple"
let str1 = $"{article} {noun}"
let str2 = Convert.ToString str1

printfn $"str1 is interned: {String.IsInterned str1 <> null}"
                  
printfn $"str1 and str2 are the same reference: {Object.ReferenceEquals(str1, str2)}"
// The example displays the following output:
//       str1 is interned: False
//       str1 and str2 are the same reference: True
// </Snippet2>

File: snippets/csharp/System/Convert/ToString/tostring_string1.cs
// <Snippet2>
using System;

public class Example
{
    public static void Main()
    {
        string article = "An";
        string noun = "apple";
        string str1 = $"{article} {noun}";
        string str2 = Convert.ToString(str1);

        Console.WriteLine($"str1 is interned: {!(string.IsInterned(str1) == null)}");
        Console.WriteLine($"str1 and str2 are the same reference: {object.ReferenceEquals(str1, str2)}");
    }
}
// The example displays the following output:
//       str1 is interned: False
//       str1 and str2 are the same reference: True
// </Snippet2>
